node/src/components/rpc_server.rs

```diff
@@ -153,7 +153,7 @@ impl RpcServer {
     ) -> Effects<Event> {
         let request = EraValidatorsRequest::new(state_root_hash.into(), protocol_version);
         effect_builder
-            .get_era_validators(request)
+            .get_era_validators_from_contract_runtime(request)
```

Comment:
Sadly there is just one situation where this call is supported by fast-sync/state pruning: getting era validators around genesis/emergency restart.

This probably is going to change the RPC interface.

CC @EdHastingsCasperLabs I'm making a follow-up ticket